collector/meminfo_numa_linux.go

```diff
@@ -69,33 +71,59 @@ func (c *meminfoNumaCollector) Update(ch chan<- prometheus.Metric) (err error) {
 		}
 		ch <- prometheus.MustNewConstMetric(desc, prometheus.GaugeValue, v, k.numaNode)
 	}
+	for k, v := range counters {
+		desc, ok := c.metricDescs[k.metricName]
+		if !ok {
+			desc = prometheus.NewDesc(
+				prometheus.BuildFQName(Namespace, memInfoNumaSubsystem, k.metricName),
+				fmt.Sprintf("Memory information field %s.", k.metricName),
+				[]string{"node"}, nil)
+			c.metricDescs[k.metricName] = desc
+		}
+		ch <- prometheus.MustNewConstMetric(desc, prometheus.CounterValue, v, k.numaNode)
+	}
 	return nil
 }
 
-func getMemInfoNuma() (map[meminfoKey]float64, error) {
-	info := make(map[meminfoKey]float64)
+func getMemInfoNuma() (map[meminfoKey]float64, map[meminfoKey]float64, error) {
+	gaugeInfo := make(map[meminfoKey]float64)
+	counterInfo := make(map[meminfoKey]float64)
 
 	nodes, err := filepath.Glob(sysFilePath("devices/system/node/node[0-9]*"))
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	for _, node := range nodes {
 		file, err := os.Open(path.Join(node, "meminfo"))
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 		defer file.Close()
 
 		numaInfo, err := parseMemInfoNuma(file)
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 		for k, v := range numaInfo {
-			info[k] = v
+			gaugeInfo[k] = v
+		}
+
+		file, err = os.Open(path.Join(node, "numastat"))
+		if err != nil {
+			return nil, nil, err
+		}
+
+		nodeNumber := nodeRE.FindStringSubmatch(node)[1]
```

Comment:
Eek - better to check the length here before accessing, otherwise it'll panic if your assumptions ever don't hold true.